Dashboards: Update config_json documentation to include k8s-style schemas (#2641)
grafana/terraform-provider-grafana: docs/resources/dashboard.md

- `config_json` (String) The complete dashboard model JSON.

> Starting with Grafana v13, use the resource corresponding to the API version they are using for Kubernetes-style dashboards.

Reviewer: ```suggestion
Starting with Grafana v13, use the resource corresponding to your dashboard's API version for Kubernetes-style dashboards.
```

> Starting with Grafana v13, it is recommended to use the "grafana_apps_dashboard_dashboard_v2" resource for Kubernetes-style dashboards.

Reviewer: ```suggestion
Starting with Grafana v13, use the `grafana_apps_dashboard_dashboard_v2` resource for Kubernetes-style dashboards.
```

Reviewer 2: We also should word this to say to use the resource corresponding to the api version they are using, not just v2

> If using this legacy resource with a v2 dashboard definition:

Reviewer: ```suggestion
If you decide to use this legacy resource with a v2 dashboard definition:
```

Reviewer 2: Can we make this generic to not just v2 (any k8s style dashboard)?

> If you decide to use this legacy resource with a k8s-style dashboard definition:

Reviewer: ```suggestion
If you decide to use this legacy resource with a Kubernetes-style dashboard definition:
```

> - In Grafana v12, only the "spec" field of the dashboard definition should be provided.

Reviewer: ```suggestion
- In Grafana v12, provide only the `spec` field of the dashboard definition.
```

> - In Grafana v13 and later, the full Kubernetes-style dashboard JSON (including "apiVersion", "kind", "metadata", and "spec") can be provided.

Reviewer: ```suggestion
- In Grafana v13 and later, provide the full Kubernetes-style dashboard JSON, including `apiVersion`, `kind`, `metadata`, and `spec`.
```
